### ISSUE TITLE
Sliding window implementation for training

### ISSUE THREAD

**Issue by Turan111** [HarryVolek/PyTorch_Speaker_Verification#12]
I haven't found sliding window implementation for training as described in the Google paper (fig.3). Only first and last 180 frames were taken from each segment. Why don't you implement sliding window? Could you explain the reason?

**Comment by HarryVolek** [comment]
A sliding window is not used during training for the paper I implemented.

**Comment by Turan111** [comment]
I am sorry, if we speak about different papers. But according to the paper "Generalized end-to-end loss for  speaker verification" fig.3  batch construction for training is shown.   
![screenshot from 2019-01-01 12-37-26](https://user-images.githubusercontent.com/6459672/50571396-7a688f00-0dc2-11e9-8482-18eda30def26.png)

**Comment by HarryVolek** [comment]
The images does not refer to the inference sliding window, but rather picking a random number of frames between 140 and 180 for each batch during training. And yes, I did not implement this, I may do it in the future.
